src/app/courses/courses.component.html

```diff
@@ -1,7 +1,7 @@
 <router-outlet></router-outlet>
 <div class="courses">
     <div class="container">
-        <app-search (searchSubmit)="onSearchSubmit($event)"></app-search>
+        <app-search (searchSubmit)="onServerSearchSubmit($event)"></app-search>
```

Comment:
I'd suggest keep it as is. On a high level it does not matter where search is performed - either in memory or on server.